### PR TITLE
Security/kube score check

### PR DESCRIPTION
**Pull request by shreyasshivakumara**
Kube-score runs over 20 different checks and some checks are not necessarily required as of now. I have added security checks that I think are necessary.

Things I have added to this PR:
1. Set `privilegedAccess` to false for all pods
2. Set `allowPrivilegeEscalation` to false for all  pods
3. Changed the `ImagePullPolicy` to Always 
4. Added headless service to statefulset deployment - Postgres and MongoDB

### CODE REVIEW

File: templates/postgresql/postgresql.yaml [hunk 1]
```diff
@@ -51,13 +51,13 @@ spec:
                 sed -i -e s,/var/lib/postgresql/10,/postgres-data,
                 /tmp/pgconf/postgresql.conf &&
                 /usr/lib/postgresql/10/bin/postgres -D /postgres-data/main
-                -c config_file=/tmp/pgconf/postgresql.conf
+                -c config_file=/etc/postgresql/10/main/postgresql.conf
```
Comment (pontus) [inline templates/postgresql/postgresql.yaml]: I believe this will break postgres persistance again, why the change?

File: templates/iip/iip-cyto.yaml [hunk 2]
```diff
@@ -47,7 +51,8 @@ spec:
           command: ["/bin/sh"]
           args: ["-c", "cp /config-temp/nginx.conf.sample /tmp/nginx.conf.sample && /tmp/deploy.sh"]
           securityContext:
-            privileged: true
+            allowPrivilegeEscalation: {{ .Values.containerSecurity.allowPrivilegeEscalation }}
+            privileged: {{ .Values.containerSecurity.privileged }}
```
Comment (pontus) [inline templates/iip/iip-cyto.yaml:51]: I don't see why these should be configurable.

File: values.yaml [hunk 3]
```diff
@@ -248,3 +248,19 @@ serviceAccount:
...
+  runAsGroup: 2000  # run as group
+  fsGroup: 1000 #file system group ID
+  capabilities:  # drop all extra capabilities
+    drop:
+      - all
+
+podSecurity:
+  runAsUser: 10000
+  fsUserGroup: 30000
+  runAsGroup: 20000
```
Comment (pontus) [inline values.yaml]: Several unused fields and I'd argue configurables for things that should not be. I also think - if they should be configurable - that they should be more granular.

**Comment by pontus** [comment]
We should also bump the chart version (on all updates).

File: templates/ims/ims.yaml [hunk 4]
```diff
@@ -41,20 +41,23 @@ spec:
...
+          #   httpGet:
+          #     path: /
+          #     port: 8080
+          # readinessProbe:
+          #   failureThreshold: 4
+          #   initialDelaySeconds: 150
+          #   periodSeconds: 10
+          #   httpGet:
+          #     path: /
+          #     port: 8080
```
Comment (pontus) [inline templates/ims/ims.yaml]: We have revision control, no need to save commented out things.

Also, we want the `readinessProbe` at least.

Comment (shreyasshivakumara) [reply templates/ims/ims.yaml]: It takes a really long time after adding ims `readinessprobe`. tbh, I don't know the reason for it. however, my next commit will include the ims `readinessProbe`. I am testing it now and it's taking a lot of time to start the core. :/

Comment (pontus) [reply templates/ims/ims.yaml]: Yes, we might need `initialDelaySeconds` of `300` or `600`.

**Review by pontus — APPROVED** [review]
